frontend/src/components/dashboard/demo/create-demo-modal/CreateDemoModal.tsx

```diff
@@ -0,0 +1,157 @@
+const CreateDemoModal = ({
+  numDemosSms,
+  numDemosTelegram,
+}: {
+  numDemosSms: number
+  numDemosTelegram: number
+}) => {
+  const modalContext = useContext(ModalContext)
+  const history = useHistory()
+  const [errorMessage, setErrorMessage] = useState('')
+  const [selectedChannel, setSelectedChannel] = useState(ChannelType.SMS)
+  const [selectedName, setSelectedName] = useState('')
+  const [demoCampaignMessage, setDemoCampaignMessage] = useState('')
+  const [isCreateEnabled, setIsCreateEnabled] = useState(true)
+  useEffect(() => {
+    // Handle case where one of the channels does not have any demos left
+    // Set the default selected channel to the channel that still has demos left
+    if (numDemosSms) {
+      setSelectedChannel(ChannelType.SMS)
+    } else {
+      setSelectedChannel(ChannelType.Telegram)
+    }
+  }, [numDemosSms])
+
+  useEffect(() => {
+    let numDemosChannel
+    let message
+    switch (selectedChannel) {
+      case ChannelType.SMS:
+        numDemosChannel = numDemosSms
+        message = `You have ${numDemosChannel}/3 SMS demo campaigns left`
+        break
+      case ChannelType.Telegram:
+        numDemosChannel = numDemosTelegram
+        message = `You have ${numDemosChannel}/3 Telegram demo campaigns left`
+        break
+      default:
+        message = `Demo campaigns are not supported for campaign type ${selectedChannel} `
+    }
+    setSelectedName(
+      `${selectedChannel} Demo ${
+        (numDemosChannel && 4 - numDemosChannel) || ''
+      }`
+    )
+    setDemoCampaignMessage(message)
+    setIsCreateEnabled(!!numDemosChannel)
+  }, [numDemosSms, numDemosTelegram, selectedChannel])
+
+  async function createDemo() {
+    try {
+      const campaign: Campaign = await createCampaign({
+        name: selectedName,
+        type: selectedChannel,
+        demoMessageLimit: 20,
+      })
+      // close modal and go to create view
+      modalContext.setModalContent(null)
+      history.push(`/campaigns/${campaign.id}`)
+    } catch (err) {
+      setErrorMessage(err.message)
+    }
+  }
+
+  return (
+    <>
+      <div className={styles.modalHeader}>{demoCampaignMessage}</div>
+      <div className={styles.section}>
+        <h5 className={styles.subtitle}>
+          The demo campaign has been named for you
+        </h5>
+        <TextInput
+          className={styles.input}
+          type="text"
+          value={selectedName}
+          disabled
+        ></TextInput>
+      </div>
+      <div className={cx(styles.section, styles.separator)}>
+        <h2 className={styles.title}>Choose the channel you want to send in</h2>
+        <h5 className={styles.subtitle}>
+          Demo campaigns are for SMS and Telegram channels only, as the Email
+          channel is already free. You may send messages to at most 20
+          recipients in each campaign. One try will be used up only after a demo
+          campaign has been sent, otherwise, it will stay as a draft.
```

Comment:
Sorry grammar nitpick - The comma after otherwise sounds little off to me, but not super sure which is correct

Reply:
The copy itself is funny, but I think whether you use a comma after "otherwise" could depend on whether you're in the Oxford comma camp. I definitely am.